python/tests/voice/test_pipecat_recv_loop_surfacing.py

```
@@ -0,0 +1,351 @@
+    Mirrors the #498 failure mode:
+      - _recv_loop delivers one 100ms batch of audio (turn 1), then the bot
+        closes the WebSocket cleanly.
+      - First recv_audio(timeout=...) returns a non-empty AudioChunk.
+      - Second recv_audio(timeout=30.0) raises PipecatRecvError fast (not
+        after a 30s hang).
+
+    The 5s asyncio.wait_for guard on the SECOND recv_audio is the speed proof:
+    if the impl had not been fixed it would block the full 30s while the
+    inbound queue is silent, then raise a bare asyncio.TimeoutError.
+    """
+    # 100ms of µ-law silence: 8000 samples/s × 0.1s = 800 bytes.
```

Ruff (0.15.17)

[warning] Comment contains ambiguous `×` (MULTIPLICATION SIGN). Did you mean `x` (LATIN SMALL LETTER X)?

(RUF003)